app/src/main/java/com/simply/birthdayapp/data/repositories/HomeRepository.kt

```diff
@@ -0,0 +1,30 @@
+package com.simply.birthdayapp.data.repositories
+
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.exception.ApolloException
+import com.simply.birthdayapp.BirthdayQuery
+import com.simply.birthdayapp.data.mappers.toBirthday
+import com.simply.birthdayapp.presentation.models.Birthday
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+interface HomeRepository {
+    suspend fun getBirthdays(): Result<List<Birthday>>
+}
+
+class HomeRepositoryImpl(
+    private val apolloClient: ApolloClient,
+) : HomeRepository {
+    override suspend fun getBirthdays(): Result<List<Birthday>> = withContext(Dispatchers.IO) {
+        val response = try {
```

Comment:
I think we don't need to have try catch block here.

Reply:
Or you can  return response as a Flow<Result<...> an handle by on each{} and catch {} blocks of flow